src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -182,16 +186,40 @@ class ContentSidebar extends PureComponent<Props, State> {
      */
     componentWillReceiveProps(nextProps: Props): void {
         const { fileId, token }: Props = this.props;
-        const { fileId: newFileId, token: newToken, hasVersions }: Props = nextProps;
+        const { fileId: newFileId, token: newToken }: Props = nextProps;
 
         const hasTokenChanged = newToken !== token;
         const hasFileIdChanged = newFileId !== fileId;
         const currentFileId = newFileId || fileId;
 
         if (currentFileId && (hasTokenChanged || hasFileIdChanged)) {
-            this.fetchFile(currentFileId);
-            if (hasVersions) {
-                this.fetchVersions(currentFileId);
+            this.fetchData(nextProps, hasFileIdChanged);
+        }
+    }
+
+    /**
+     * Fetches the data for the sidebar
+     *
+     * @param {Object} Props the component props
+     * @param {boolean} hasFileIdChanged true if the file id has changed
+     */
+    fetchData({ fileId, hasVersions, hasActivityFeed, hasAccessStats }: Props, hasFileIdChanged?: boolean) {
+        if (hasFileIdChanged) {
```

Comment:
This is not needed for componentDidMount(), so should move it inside componentWillMount()
I am assuming you are adding this for the loading indicator to show.

Reply:
Did you mean in ```componentWillReceiveProps``` ? I added this so that the sidebar wouldn't have stale data for things such as comments when the current fileId changes